### ISSUE TITLE
Error - failed to get runspace

### ISSUE THREAD

**Issue by bjornvdh** [ironmansoftware/powershell-universal#1665]
### Steps to Reproduce

Every 1 or 2 days, Powershell Universal hangs.
In older versions, it would just hang, without any response, but in the current version, all users get the error message “Failed to get runspace.”.

Is there any way to prevent this?

It appears to occur more often on days where the dashboards get used more. I don't have an exact method of reproducing this, so:

1) Run a dashboard that interacts with Active Directory, SQL server, TopDesk API and some other API's.
2) Get a bunch of users to use it on a regular basis.

### Expected behavior

```console
The system remains responsive and useable.
```

### Actual behavior

```console
At random time intervals, the dashboards stop responding.
The user gets an error "Failed to get runspace.".
```

### Environment data

Windows Server 2019
Powershell Universal 3.4.1 installed through MSI.
ADFS for authentication / single sign on through WS-FED.
Powershell 7.2.2 (x64) - The dashboard now runs on the integrated environment; which results in the same behavior.

### Visuals

![afbeelding](https://user-images.githubusercontent.com/80324538/199240399-13ed2277-83a8-4541-bef4-1d4522afb05e.png)

**Comment by adamdriscoll** [comment]
This issue has been mentioned on **Ironman Software Forums**. There might be relevant details there:

https://forums.ironmansoftware.com/t/failed-to-get-runspace/8055/2

**Comment by adamdriscoll** [comment]
This is typically a performance issue. Can you check the CPU and Memory usage of PowerShell Universal? The error refers to attempting to get a runspace from the runspace pool but it doesn't happen within the 5 second timeout. This could be because they are all currently being used or a large number of them are locked. 

You should also be able to click on the Processes card on the home page. 

<img width="201" alt="image" src="https://user-images.githubusercontent.com/1256531/199279918-f79c0b00-bb3e-4feb-b6f3-09423fd754e4.png">

From there, you'll be able to see the status of the runspaces in the process. 

<img width="559" alt="image" src="https://user-images.githubusercontent.com/1256531/199280001-19923611-e056-4f54-a56c-571e67bca599.png">

 The integrated environment may be exasperating it because it can't be restarted as easily as an external dashboard process. 

In 3.5, we're looking at providing dashboard and API recycling to automatically detect certain conditions and restart the dashboard processes.

**Comment by adamdriscoll** [comment]
I would also recommend increasing the max runspace count for the environment.

**Comment by bjornvdh** [comment]
Thanks for your quick reply, as always!
I tried it right after you posted and Neo as well on the forums.
It seems to be running stable now. It is weird though, because when I look, at the processes stat on the dashboard, it rarely gets above 15.
And, when I got the “Failed to get runspace” error, it never seemed to recover, until i restarted PSU. So it seems like sometimes a runspace becomes invalid or something, and then is blocked. 

Anyway, I’ll keep monitoring it for now. 
Would it be possible to increase the default max runspaces, or at least make a note of this in de documentation?
